### PR TITLE
[Feature] file parser option for FilePicker widget [csv support]

### LINKED ISSUE #1776: Add option to  parse file content - File Picker widget
### Description
Parse the selected files, with support for 

- [ ] csv
- [ ] excel

### PR DESCRIPTION
**Pull request by arpitnath**
Resolves #1776 [support: csv]
- [x] CSV file parser 
- [x] Add documentation

### CODE REVIEW

**Comment by arpitnath** [comment]
NOTE: just included `PDF`  option for testing, will remove that

File: frontend/src/Editor/Components/FilePicker.jsx [hunk 1]
```diff
@@ -127,12 +129,19 @@ export const FilePicker = ({ width, height, component, currentState, onComponent
 
     // * readAsDataURL
     const readFileAsDataURL = await getFileData(file, 'readAsDataURL');
+    const autoDetectFileType = fileTypeFromExtension === 'auto-detect' ? true : false;
```
Comment (gondar00) [inline frontend/src/Editor/Components/FilePicker.jsx:132]:
```suggestion
    const autoDetectFileType = fileTypeFromExtension === 'auto-detect';
```

File: frontend/src/Editor/Components/FilePicker.jsx [hunk 2]
```diff
@@ -281,3 +290,47 @@ FilePicker.AcceptedFiles = ({ children, width, height, showFilezone, bgThemeColo
...
+  const newArray = rows.map((row) => {
+    const values = row.split(delimiter);
+    const eachObject = headers.reduce((obj, header, i) => {
+      obj[header] = values[i];
+      return obj;
+    }, {});
+    return eachObject;
+  });
+
+  return newArray;
```
Comment (gondar00) [inline frontend/src/Editor/Components/FilePicker.jsx:307]: probably can add a try catch here to handle bad data in csv

**Review by gondar00 — CHANGES_REQUESTED** [review]

**Review by gondar00 — DISMISSED** [review]

**Review by sherfin94 — CHANGES_REQUESTED** [review]
@arpitnath It works awesome!. Can we also add an `isParsing` property so that I can show loading icon on table when the file is still being processed?

**Comment by arpitnath** [comment]
> @arpitnath It works awesome!. Can we also add an `isParsing` property so that I can show loading icon on table when the file is still being processed?

Yeah sure 😎

**Review by Navaneeth-pk — APPROVED** [review]

**Pull request merged**